clients/spark/src/main/scala/io/treeverse/clients/ApiClient.scala

```diff
@@ -168,7 +168,7 @@ class ApiClient private (conf: APIConfigurations) {
     val storageNamespace = key.storageClientType match {
       case StorageClientType.HadoopFS =>
         ApiClient
-          .translateURI(URI.create(repo.getStorageNamespace), getBlockstoreType)
+          .translateURI(URI.create(repo.getStorageNamespace), getBlockstoreType(repo.getStorageId))
```

Comment:
The storage namespace information is cached in this client.
The blockstore type is not cached and it means that we call the server multiple times.
Suggest to add this information to the cache or cache the configucation at the same level.

Reply:
was looking at the namespace cache and missed the key to namespace cache - can ignore my comment.